virtool/settings/data.py

```diff
@@ -0,0 +1,57 @@
+from virtool_core.models.settings import Settings
+
+from virtool.mongo.core import DB
+from virtool.settings.oas import UpdateSettingsSchema
+
+PROJECTION = {"_id": False}
+
+
+class SettingsData:
+    def __init__(self, db):
+        self._db: DB = db
+
+    async def get_all(self) -> Settings:
+        """
+        List all settings.
+
+        :return: the application settings
+
+        """
+
+        settings = await self._db.settings.find_one(
+            {"_id": "settings"}, projection=PROJECTION
+        )
+
+        if settings:
+            return Settings(**settings)
```

Comment:
Hmm. I would think we should raise an exception if the settings are not defined. It should never happen.

Reply:
I believe this only occurs in test_yaml_loading.py because the app fixture does not have a data layer and ensure is not being called. I could add them directly to the fixture or I could do this as a workaround: 

async def test_load_yml(app, data_layer, fake, example_test_case):

    app["data"] = data_layer

    await get_data_from_app(app).settings.ensure()

then it would work as well.